lib/pbio/drv/ioport/ioport_lpf2.c

```diff
@@ -481,12 +481,26 @@ PROCESS_THREAD(pbdrv_ioport_lpf2_process, ev, data) {
     }
 
     while (true) {
-        PROCESS_WAIT_EVENT_UNTIL(ev == PROCESS_EVENT_TIMER && etimer_expired(&timer));
-        etimer_reset(&timer);
+        PROCESS_WAIT_EVENT();
 
+        // Reset timer on timer event
+        if (ev == PROCESS_EVENT_TIMER && etimer_expired(&timer)) {
+            etimer_reset(&timer);
+        }
+
+        // Update all devices, either on timer or on reset
         for (int i = 0; i < PBDRV_CONFIG_IOPORT_LPF2_NUM_PORTS; i++) {
             ioport_dev_t *ioport = &ioport_devs[i];
 
+            // If pbio_uartdev_process tells us the uart device was removed, reset
+            // ioport id so the next timer event will take care of resetting the ioport
+            if (ev == PROCESS_EVENT_SERVICE_REMOVED) {
+                if (ioport->iodev == (pbio_iodev_t *)data) {
+                    ioport->connected_type_id = PBIO_IODEV_TYPE_ID_NONE;
+                }
+                continue;
+            }
+
```

Comment:
`poll_dcm(ioport)` must be called at 2 ms intervals to work correctly. So we need to add something to ensure that it is only called when the timer event was received.

Reply:
That's what this is doing, right?

Reply:
It looks like the timer is reset is guarded by `if (ev == PROCESS_EVENT_TIMER && etimer_expired(&timer))` but the call to `poll_dcm(ioport)` is outside of that, so it will be called when any event other than `PROCESS_EVENT_SERVICE_REMOVED` happens. There are other events that broadcast to all processes.